src/imagej/images.py

```diff
@@ -0,0 +1,29 @@
+"""
+Utility functions for creating and working with images.
+"""
+
+
+def is_arraylike(arr):
+    return (
+        hasattr(arr, "shape")
+        and hasattr(arr, "dtype")
+        and hasattr(arr, "__array__")
+        and hasattr(arr, "ndim")
+    )
+
+
+def is_memoryarraylike(arr):
+    return (
+        is_arraylike(arr)
+        and hasattr(arr, "data")
+        and type(arr.data).__name__ == "memoryview"
+    )
+
+
+def is_xarraylike(xarr):
+    return (
+        hasattr(xarr, "values")
+        and hasattr(xarr, "dims")
+        and hasattr(xarr, "coords")
+        and is_arraylike(xarr.values)
+    )
```

Comment:
Why not scyjava? Since it's just duck-type checking, it could live in scyjava without incurring a dependency on numpy and/or xarray. My perspective here is that people are starting to use scyjava who don't care about ImgLib2, but might care about numpy on the Python side, since it's so ubiquitous. It's even tempting to move the Java primitive array wrapping logic into scyjava somehow, although it is tricky dependency-wise.

Reply:
I'm looking at scyjava. Sections include:

* "Type conversion utilities" - Priority and Converter classes
* "Python to Java" - `isjava`, `jclass`, `jimport`, `jstacktrace`, `add_java_converter`, `to_java`
* "Utility functions" - `get_version`, `is_version_at_least`, `compare_version`

Where do you think these three new `is_` functions belong? One of the above three? Or a new section perhaps?

Reply:
I suppose either "Type conversion utilities", or "Python to Java" would be suitable?